src/renderer/components/dock/terminal/send-command.injectable.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable, lifecycleEnum } from "@ogre-tools/injectable";
+import { when } from "mobx";
+import { TerminalApi, TerminalChannels } from "../../../api/terminal-api";
+import { bind, noop } from "../../../utils";
+import { Notifications } from "../../notifications";
+import selectDockTabInjectable from "../dock/select-dock-tab.injectable";
+import type { DockTab, TabId } from "../dock/store";
+import createTerminalTabInjectable from "./create-terminal-tab.injectable";
+import getTerminalApiInjectable from "./get-terminal-api.injectable";
+
+interface Dependencies {
+  selectTab: (tabId: TabId) => void;
+  createTerminalTab: () => DockTab;
+  getTerminalApi: (tabId: TabId) => TerminalApi;
+}
+
+export interface SendCommandOptions {
+  /**
+   * Emit an enter after the command
+   */
+  enter?: boolean;
+
+  /**
+   * @deprecated This option is ignored and infered to be `true` if `tabId` is not provided
+   */
+  newTab?: any;
+
+  /**
+   * Specify a specific terminal tab to send this command to
+   */
+  tabId?: TabId;
+}
+
+async function sendCommand({ selectTab, createTerminalTab, getTerminalApi }: Dependencies, command: string, options: SendCommandOptions = {}): Promise<void> {
+  let { tabId } = options;
+
+  if (tabId) {
+    selectTab(tabId);
+  } else {
+    tabId = createTerminalTab().id;
+  }
+
+  await when(() => Boolean(getTerminalApi(tabId)));
+
+  const terminalApi = getTerminalApi(tabId);
+  const shellIsReady = when(() =>terminalApi.isReady);
```

Comment:
Imagine if `terminalApi` was a dependency. And only present if it is ready. No need for `when`